protocol/validation/block.go

```diff
@@ -32,7 +32,7 @@ var (
 // See $CHAIN/protocol/doc/spec/validation.md#accept-block.
 // It evaluates the prevBlock's consensus program,
 // then calls ValidateBlock.
-func ValidateBlockForAccept(ctx context.Context, snapshot *state.Snapshot, prevBlock, block *bc.Block, validateTx func(*bc.Tx) error) error {
+func ValidateBlockForAccept(ctx context.Context, snapshot *state.Snapshot, prevBlock, block *bc.Block, initialBlockHash bc.Hash, validateTx func(*bc.Tx) error) error {
```

Comment:
Maybe if we put the initial block hash in the `state.Snapshot`, we can avoid putting it on the `protocol.Chain` (and changing the interface of `asset.NewRegistry`)?